opendict.1
.TH OPENDICT 1 "2017-11-20"
.\" --------------------------------------------------------
.SH NAME
opendict \- computer dictionary with several dictionary format support
.\" --------------------------------------------------------
.SH SYNOPSIS
.B opendict
.sp
or select
.B Dictionary OpenDict
from the
.B Utility
submenu of the Applications menu if using Gnome Desktop Environment.
.\" --------------------------------------------------------
.SH DESCRIPTION
.B OpenDict
is a free cross-platform dictionary program. 
It works with DICT, Slowo and Mova dictionaries. It also supports plug-in
dictionaries that may be created for almost any data source. OpenDict is a
client for DICT servers.

.\" --------------------------------------------------------
.SH SEE ALSO
.BR http://opendict.sf.net
.BR https://github.com/nerijus/opendict
.\" --------------------------------------------------------
.SH AUTHORS
.B
opendict
was written by Martynas Jocius <martynas.jocius@idiles.com>.

.ad l
This manual page was written by Kestutis Biliunas
.nh 
<kebil@kaunas.init.lt>,
.hy
for the Debian system (but may be used by others).




ChangeLog
2017-11-20 nerijus

	* strip whitespace when searching

2014-04-08 nerijus

	* PyXML is no longer needed

2006-02-15 18:29  mjoc

	* lib/gui/mainwin.py, po/lt.po, po/opendict.pot: Window "Edit
	  Dicitonaries" renamed to "Create Dictionaries".

2006-02-14 15:17  mjoc

	* po/: lt.po, opendict.pot: Translation updated.

2006-02-13 02:08  nerijus

	* lib/gui/mainwin.py: clear search field on ESC, implements feature
	  request 1351220

2006-02-12 22:10  nerijus

	* po/lt.po, win32/setup.manifest.py, win32/setup.py: Version
	  changed to 0.6.1

2006-02-12 20:59  mjoc

	* po/: lt.po, opendict.pot: Translation (po) file updated.

2006-02-12 20:56  mjoc

	* lib/gui/dictconnwin.py, po/opendict.pot: Bug fix: encoding has
	  not been really used after chaning it in DictConnection dialog.

2006-02-12 02:53  mjoc

	* ChangeLog, opendict.1, po/lt.po, po/opendict.pot: Manual page
	  added. ChangeLog updated.

2006-01-28 16:45  mjoc

	* opendict.py, lib/info.py, lib/logger.py, lib/gui/dictconnwin.py,
	  lib/gui/helpwin.py, lib/gui/mainwin.py: 1. Debug logger messages
	  minimized; 2. DICT connection window sets encoding on the main
	  window menu; 3. Version changed to 0.6.1;

2006-01-08 23:12  nerijus

	* lib/misc.py: fixed spelling

2006-01-08 14:14  mjoc

	* lib/config.py, lib/gui/dictconnwin.py, po/lt.po, po/opendict.pot:
	  DICT connection window changes: 1) New encoding selection entry
	  added. 2) Windows remembers last server and encoding used.

2005-12-26 20:04  nerijus

	* lib/: misc.py, parser.py, gui/mainwin.py: fixed grammar

2005-12-26 19:42  nerijus

	* lib/parser.py: removed duplicated line

2005-11-10 13:45  mjoc

	* lib/parser.py: If invalid line is found while parsing Slowo
	  format dictionary, error message is written to system.log. Until
	  this bug fix exception had not been cached.

2005-11-02 22:12  mjoc

	* copying.html: FSF address updated.

2005-11-02 21:58  mjoc

	* lib/: tests/test_editor.py, extra/html2text.py: First line with
	  #!/exec/path removed.

2005-10-26 00:19  nerijus

	* win32/compile.bat: copy msvcr71.dll

2005-10-25 17:42  mjoc

	* TODO.txt, opendict.py, po/opendict.pot: Very small changes: one
	  more todo item, etc.

2005-10-25 16:51  nerijus

	* win32/: setup.manifest.py, setup.py: updated version to 0.6.0

2005-10-24 22:39  mjoc

	* ChangeLog: New ChangeLog generated using cvs2cl utility.

2005-10-24 22:04  mjoc

	* lib/gui/mainwin.py, po/lt.po, po/opendict.pot: Show/Hide button
	  removed (hidden) from the main window, menu item of the same
	  functionality added. Lithuanian translation updated.

2005-10-20 18:26  mjoc

	* lib/info.py: Version changed to 0.6.0. Going to release stable
	  version 0.6.0.

2005-10-20 18:22  mjoc

	* misc/opendict.desktop: Name changed from "OpenDict" to
	  "Dictionary OpenDict". More intuitive.

2005-09-06 22:59  mjoc

	* doc/Plugin-HOWTO.html: Simple plugin example section written.
	  Needs more detailed instructions in the future.

2005-08-31 11:39  mjoc

	* doc/Plugin-HOWTO.html: Introduction, Types od OpenDict
	  Dictionaries and Plain Dictionary Example sections written.

2005-08-30 00:27  mjoc

	* doc/: OpenDict_plugin_dev.txt, Plugin-HOWTO.html,
	  Specification.txt: Old plugin-dev doc removed, spefication
	  removed, new plugin-dev howto added (not finished yet).

2005-08-17 19:46  mjoc

	* pixmaps/: icon-24x24.png, icon-32x32.png, icon-48x48.png,
	  icon-620x620.png, icon-96x96.png, SVG/icon-rune.svg: Icon
	  improved a bit by adding some color transitioning.

2005-08-06 23:37  mjoc

	* TODO.txt, lib/gui/mainwin.py, lib/gui/prefswin.py, po/lt.po,
	  po/opendict.pot: Option for default clipboard scanner use added
	  into preferences window.

2005-08-05 22:02  mjoc

	* TODO.txt, lib/info.py, lib/gui/mainwin.py, po/opendict.pot:
	  Version changed to 0.5.9-CVS, clipboard-scanned text is trimmed
	  before searching for translation. New TODO entries added.

2005-06-14 00:47  nerijus

	* README.txt, TODO.txt: updated README; converted to unix line
	  endings

2005-06-13 23:58  nerijus

	* README.txt, opendict.py, lib/info.py: improved error message when
	  PyXML is missing, so thhat Fedora users know which package they
	  are missing; bumped version to 0.5.8

2005-06-10 23:35  mjoc

	* lib/tests/data/sampleplugin/sample.py: Sample plugin for unit
	  testing updated to work with new module tree after adding 'lib'
	  to import path.

2005-06-10 23:17  mjoc

	* po/: lt.po, opendict.pot: lt.po merging conflict resolved.

2005-06-10 23:10  mjoc

	* AUTHORS.txt: Added authors file.

2005-06-10 23:04  mjoc

	* ChangeLog, lib/gui/helpwin.py, lib/gui/mainwin.py, po/lt.po,
	  po/opendict.pot: 1. Controls are enabled before showing search
	  results  2. Nerijus B. added to authors list.

2005-05-31 14:34  nerijus

	* opendict.py: ask unicode wx version; it still can use ansi
	  version if unicode is not available

2005-05-24 18:05  nerijus

	* Makefile: simplified install/uninstall

2005-05-16 21:53  mjoc

	* lib/info.py, lib/newplugin.py, lib/util.py, lib/gui/mainwin.py,
	  po/lt.po, po/opendict.pot: wx.EndBusyCursor bug fixed, no error
	  message on Windows build. Name correction is not used until
	  correct functionality is achieved.

2005-04-29 03:18  nerijus

	* win32/setup.py: no changes

2005-04-29 03:16  nerijus

	* win32/: compile.bat, setup.manifest.py: added setup script with
	  manifest

2005-04-29 02:27  nerijus

	* win32/compile.bat: copy license file manually

2005-04-29 02:09  nerijus

	* win32/compile.bat: encodings work with py2exe

2005-04-28 06:03  nerijus

	* lib/info.py: py2exe now works

2005-04-28 05:58  nerijus

	* opendict.py, lib/info.py, win32/compile.bat, win32/setup.py:
	  py2exe now works

2005-04-27 18:56  mjoc

	* opendict.py: wxPython 2.5 is selected only if application is not
	  frozen (i.e. compiled using py2exe)

2005-04-27 18:46  mjoc

	* opendict.py, lib/enc.py: OpenDict now supports wxPython
	  multiversioning. If both wxPython 2.4 and 2.5 are installed on

opendict.py
#!/usr/bin/env python
# -*- coding: iso-8859-1 -*-

# OpenDict
# Copyright (c) 2003-2006 Martynas Jocius <martynas.jocius@idiles.com>
# Copyright (c) 2007 IDILES SYSTEMS, UAB <support@idiles.com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your opinion) any later version.
#
# This program is distributed in the hope that will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MECHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more detals.
#
# You shoud have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
# 02110-1301 USA

import sys
import os
import imp
import traceback
import string
import time

# main_is_frozen() returns True when running the exe, and False when
# running from a script.
def main_is_frozen():
    return (hasattr(sys, "frozen") or # new py2exe
	    hasattr(sys, "importers") # old py2exe
	    or imp.is_frozen("__main__")) # tools/freeze

# If application is not frozen to binary, try selecting wxPython 3.0 or 2.8
# on multiversioned wxPython installation.
if not main_is_frozen():
    try:
        import wxversion
        wxversion.select(["2.8-unicode", "3.0"])
    except Exception, e:
        print "You seem to have an unsupported wxPython version: %s" \
              % e

try:
    import wx
except ImportError:
    print >> sys.stderr, "**"
    print >> sys.stderr, "** Error: wxPython library not found"
    print >> sys.stderr, "** Please install wxPython 2.8 or later to run OpenDict"
    print >> sys.stderr, "**"
    sys.exit(1)


# get_main_dir() returns the directory name of the script or the
# directory name of the exe
def get_main_dir():
    if main_is_frozen():
	return os.path.dirname(sys.executable)
    return os.path.dirname(os.path.realpath(__file__))
    # or return os.path.dirname(sys.argv[0])

#
# Initial path
#
sys.path.insert(0, get_main_dir())

# OpenDict Modules
from lib import info
from lib.gui.mainwin import MainWindow
from lib.gui.errorwin import ErrorWindow
from lib.config import Configuration
from lib.logger import systemLog, debugLog, DEBUG, INFO, WARNING, ERROR
from lib import misc
from lib import info
from lib import newplugin
from lib import plaindict
from lib import util


class OpenDictApp(wx.App):
   """Top-level class of wxWidgets application"""

   locale = wx.Locale()

   def OnInit(self):

      _ = wx.GetTranslation
      _start = time.time()

      wx.Version = []
      try:
          wx.Version = wx.__version__
      except Exception, e:
          try:
              wx.Version = wx.Python.__version__
          except:
              pass

      if wx.Version.split('.') < ['2', '8']:
          from lib.gui import errorwin

          title = _("wxPython Version Error")
          msg = _("wxPython %s is installed on this system.\n\n"
                  "OpenDict %s requires wxPython 2.8 or newer to run smoothly.\n\n"
                  "You can find wxPython at "
                  "http://www.wxpython.org or you can "
                  "install it using your system package manager.") \
                  % (wx.Version, info.VERSION)
          errorwin.showErrorMessage(title, msg)
          return False


      util.makeDirectories()

      systemLog(DEBUG, "Unicode version: %s" % wx.USE_UNICODE)

      # Init gettext support
      wx.Locale_AddCatalogLookupPathPrefix(os.path.join(info.GLOBAL_HOME,
                                                       'po'))
      self.locale.Init(wx.LANGUAGE_DEFAULT)
      self.locale.AddCatalog('opendict')

      # Data cache instance
      self.cache = {}

      # Dictionaries container
      # Mapping: name -> object
      self.dictionaries = {}

      # Failed dictionaries.
      # For error message that may be shown after creating main window
      self.invalidDictionaries = []

      self.config = Configuration()
      self.config.load()

      self.agreements = util.AgreementsManager(os.path.join(info.LOCAL_HOME,
                                                            'agreements.txt'))



      # Set unique ids
      for plugin in newplugin.loadDictionaryPlugins(self.dictionaries,
                                                    self.invalidDictionaries):
         self.config.ids[wx.NewId()] = plugin.getName()

      for plain in plaindict.loadPlainDictionaries(self.dictionaries):
         self.config.ids[wx.NewId()] = plain.getName()


      for d in self.dictionaries.values():
          if not self.config.activedict.init:
              if not self.config.activedict.enabled(d.getName()):
                  d.setActive(active=False)
          else:
              # Fill up with names if not initialized yet
              self.config.activedict.add(d.getName())


      windowPos = (int(self.config.get('windowPosX')),
                                int(self.config.get('windowPosY')))
      windowSize = (int(self.config.get('windowWidth')),
                    int(self.config.get('windowHeight')))

      self.window = MainWindow(None, -1, "OpenDict",
                               windowPos,
                               windowSize,
                               style=wx.DEFAULT_FRAME_STYLE)

      try:
          systemLog(INFO, "OpenDict %s" % info.VERSION)
          systemLog(INFO, "wxPython %s" % wx.Version)
          systemLog(INFO, "Global home: %s:" % info.GLOBAL_HOME)
          systemLog(INFO, "Local home: %s" % info.LOCAL_HOME)
          systemLog(DEBUG, "Loaded in %f seconds" % (time.time() - _start))
      except Exception, e:
          print "Logger Error: Unable to write to log (%s)" % e

      self.window.Show(True)

      return True


if __name__ == "__main__":

   openDictApp = OpenDictApp(0)
   openDictApp.MainLoop()
